skglm/estimators.py

```diff
@@ -285,7 +241,9 @@ def fit(self, X, y):
         """
         self.penalty = self.penalty if self.penalty else L1(1.)
         self.datafit = self.datafit if self.datafit else Quadratic()
-        return _glm_fit(X, y, self, self.datafit, self.penalty)
+        self.solver = self.solver if self.solver else AcceleratedCD()
+        return self.solver.solve(X, y, self.datafit, self.penalty)
```

Comment:
you need glm_fit here to do all the checks, and if solver is None this will fail so you need a default instantiation like for penalty and datafit